test/uk/gov/hmrc/timetopayproxy/connectors/TtpConnectorSpec.scala

```diff
@@ -119,6 +119,51 @@ class TtpConnectorSpec extends PlaySpec with DefaultAwaitTimeout with FutureAwai
     }
 
     "using TTP" must {
+      "return a InvalidMainAndOrSubTransTypes error that TTP-proxy passes upstream" in new Setup(false) {
+        stubPostWithResponseBody(
+          "/debts/time-to-pay/quote",
+          400,
+          errorResponse(
+            "BAD_REQUEST",
+            "Invalid MainTransTypes and/or SubTransTypes. MainTransTypes: 1111 SubTransTypes:  values do not match stored charge types"
+          )
+        )
+        val result = connector.generateQuote(
+          GenerateQuoteRequest(
+            CustomerReference("CustRef1234"),
+            ChannelIdentifier.Advisor,
+            PlanToGenerateQuote(
+              QuoteType.Duration,
```

Comment:
Shall we use named arguments here? Not a big deal, but letting the next dev know what each 'None' is relating to for example is a big help in speed :)